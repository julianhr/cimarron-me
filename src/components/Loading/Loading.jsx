import React from 'react'
import PropTypes from 'prop-types'
import { css, keyframes } from '@emotion/core'
import styled from '@emotion/styled'


const Root = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 70vh;
`

const Container = styled.div`
  display: flex;
  font-size: 24px;
`

const bounceDot = keyframes`
  0%, 14% {
    transform: translateY(0);
  }

  7% {
    transform: translateY(-14px);
  }
`

const spanBase = css`
  font-size: 34px;
  padding: 0 3px;
  margin-top: -7px;
`

const Dot1 = styled.span`
  ${spanBase}
  animation: ${bounceDot} 1400ms ease-in 0s infinite;
`

const Dot2 = styled.span`
  ${spanBase}
  animation: ${bounceDot} 1400ms ease-in 280ms infinite;
`

const Dot3 = styled.span`
  ${spanBase}
  animation: ${bounceDot} 1400ms ease-in 560ms infinite;
`

function Loading() {
  return (
    <Root>
      <Container>
        <span>Loading</span><Dot1>.</Dot1><Dot2>.</Dot2><Dot3>.</Dot3>
      </Container>
    </Root>
  )
}

export default Loading
