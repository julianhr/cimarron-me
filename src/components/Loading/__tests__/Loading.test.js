import React from 'react'
import { render } from 'enzyme'
import emotionSerializer from 'jest-emotion'

import Loading from '../Loading'


describe('Loading', () => {
  it('matches snapshot', () => {
    const wrapper = render(<Loading />)
    expect.addSnapshotSerializer(emotionSerializer)
    expect(wrapper).toMatchSnapshot()
  })
})
