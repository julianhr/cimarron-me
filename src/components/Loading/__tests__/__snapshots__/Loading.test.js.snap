// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading matches snapshot 1`] = `
@keyframes animation-0 {
  0%, 14% {
    -webkit-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }

  7% {
    -webkit-transform: translateY(-14px);
    -ms-transform: translateY(-14px);
    transform: translateY(-14px);
  }
}

@keyframes animation-0 {
  0%, 14% {
    -webkit-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }

  7% {
    -webkit-transform: translateY(-14px);
    -ms-transform: translateY(-14px);
    transform: translateY(-14px);
  }
}

@keyframes animation-0 {
  0%, 14% {
    -webkit-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }

  7% {
    -webkit-transform: translateY(-14px);
    -ms-transform: translateY(-14px);
    transform: translateY(-14px);
  }
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 70vh;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 24px;
}

.emotion-0 {
  font-size: 34px;
  padding: 0 3px;
  margin-top: -7px;
  -webkit-animation: animation-0 1400ms ease-in 0s infinite;
  animation: animation-0 1400ms ease-in 0s infinite;
}

.emotion-2 {
  font-size: 34px;
  padding: 0 3px;
  margin-top: -7px;
  -webkit-animation: animation-0 1400ms ease-in 280ms infinite;
  animation: animation-0 1400ms ease-in 280ms infinite;
}

.emotion-4 {
  font-size: 34px;
  padding: 0 3px;
  margin-top: -7px;
  -webkit-animation: animation-0 1400ms ease-in 560ms infinite;
  animation: animation-0 1400ms ease-in 560ms infinite;
}

<div
  class="emotion-8 emotion-9"
>
  <div
    class="emotion-6 emotion-7"
  >
    <span>
      Loading
    </span>
    <span
      class="emotion-0 emotion-1"
    >
      .
    </span>
    <span
      class="emotion-2 emotion-3"
    >
      .
    </span>
    <span
      class="emotion-4 emotion-5"
    >
      .
    </span>
  </div>
</div>
`;
